src/middleware/devtools.ts

```diff
@@ -72,11 +70,12 @@ type StoreDevtools<S> = S extends {
     }
   : never
 
-export interface DevtoolsOptions extends Config {
+export interface DevtoolsOptions extends Omit<Config, 'actionCreators'> {
   name?: string
   enabled?: boolean
   anonymousActionType?: string
   store?: string
+  actionCreators?: boolean | (<A, P extends any[] = any[]>(...args: P) => A) | Record<string, boolean |(<A, P extends any[] = any[]>(...args: P) => A)>;
```

Comment:
Zustand philosophy is primitive, simple, small.

I think this should be simplified much further.

Reply:
🤔 
```ts
actionCreators?: boolean | { [dotNotationProp: string]: boolean } | ActionCreatorDefinition[]
```
or
```ts
actionCreators?: 'top-level-funcs' | 'all-nested-funcs' | ActionCreatorDefinition[]
```

Reply:
Hmm, I was cleaning up the code and was doing like this:
```ts

type ActionCreator = <A, P extends any[] = any[]>(...args: P) => A;
type ActionCreatorMask = {
  [key: string]: boolean | ActionCreator | ActionCreatorMask;
};

export interface DevtoolsOptions extends Omit<Config, 'actionCreators'> {
  /* ... */
  actionCreators?: boolean | ActionCreatorMask;
}
```

Still too complex? Idea was to allow passing a mask of functions to expose, dot notation could work.

Reply:
I lean more towards the first version you mentioned